src/librustc_resolve/lib.rs

```diff
@@ -3010,6 +3010,20 @@ impl<'a, 'crateloader: 'a> Resolver<'a, 'crateloader> {
             let code = DiagnosticId::Error(code.into());
             let mut err = this.session.struct_span_err_with_code(base_span, &base_msg, code);
 
+            // Emit help message for fake-self from other languages like `this`(javascript)
+            let fake_self: Vec<Ident> = ["this", "my"].iter().map(
+                |s| Ident::from_str(*s)
+            ).collect();
+            if fake_self.contains(&item_str)
+                && this.self_value_is_available(path[0].span, span) {
```

Comment:
Can we ever get here in case that `this` or `my` are defiend? If so, maybe we should add a check here to only suggest this when they are undefined and the author actually meant what we call `self`?

A test case to consider adding would be

```rust
struct Foo { x: i32 };
impl Foo {
    fn foo(&mut self) -> i32 {
        let this = 13;
        // ... a hundred complicated lines of code ...
        this.x
    }
}
```

Reply:
yes, it's covered. I tested this locally(but deleted this lines from test case):
```rust
let this = self;
this.x
```
it works fine and doesn't  trigger the err.
I'll recover this as you point here.